torch/ao/nn/sparse/linear.py

```diff
@@ -0,0 +1,32 @@
+import torch
+from torch.nn import functional as F
+
+class Linear(torch.nn.Linear):
```

Comment:
We should have the masking done by a module that can operate on the weights. This would be a FakeSparseModule, whos forward method masks the weights.  Can you rewrite this implementation such that:
1. We have a base FakeSparseModule class that contains a mask attribute.
2. The FakeSparseModule is applied to the weight as a parameterization.

Reply:
Sure, in this PR lets do the following:
1. define fakesparse module [similar to the mulby module]
```python 
class FakeSparse(nn.Module):
    def __init__(self, shape):
        super().__init__()
       # Initialize mask to be all ones

        self.mask = torch.ones(shape)

    def forward(self, x):
        assert self.mask.shape == x.shape
        return self.mask * x
```

If you are doing the reparameterization in a later PR, use this module manually, similar to how it is used in nn.qat.linear.
```
 def __init__(self, in_features, out_features, bias=True, device=None, dtype=None) -> None:
        factory_kwargs = {'device': device, 'dtype': dtype}
        super().__init__(in_features, out_features, bias, **factory_kwargs)
        self.fake_sparse = FakeSparse(self.weight.size())

    def forward(self, input):
        return F.linear(input, self.fake_sparse(self.weight), self.bias)
```